core/sub.mk

```diff
@@ -10,3 +10,17 @@ recipe-ta_pub_key = scripts/pem_to_pub_c.py --prefix ta_pub_key \
 		--key $(TA_SIGN_KEY) --out $(sub-dir-out)/ta_pub_key.c
 cleanfiles += $(sub-dir-out)/ta_pub_key.c
 endif
+
+ifeq ($(CFG_WITH_USER_TA)-$(CFG_EARLY_TA),y-y)
+define process_early_ta
+early-ta-$1-uuid := $(firstword $(subst ., ,$(notdir $1)))
+gensrcs-y += early-ta-$1
+produce-early-ta-$1 = early_ta_$$(early-ta-$1-uuid).c
+depends-early-ta-$1 = $1 scripts/ta_bin_to_c.py
+recipe-early-ta-$1 = scripts/ta_bin_to_c.py --ta $1 \
+		--out $(sub-dir-out)/early_ta_$$(early-ta-$1-uuid).c
+cleanfiles += $(sub-dir-out)/early_ta_$$(early-ta-$1-uuid).c
+cflags-remove-early-ta-$1-y := -pedantic
```

Comment:
Putting a `__extension__` in front of the `const struct early_ta ...` would eliminate the need of removing the `-pedantic` flag. It relies on a compiler extension after all.